packages/@react-aria/dnd/src/useDroppableCollection.ts

```diff
@@ -49,6 +50,52 @@ export function useDroppableCollection(props: DroppableCollectionOptions, state:
   localState.props = props;
   localState.state = state;
 
+  let {
+    onInsert,
+    onRootDrop,
+    onItemDrop,
+    onReorder
+  } = localState.props;
+
+  useEffect(() => {
+    // Update the global DnD state via a capturing drop listener since the onDrop passed
+    // to useDrop happens after onDragEnd (mouse drag and drop only). onDragEnd needs droppedCollectionRef
+    // and droppedTarget to be set so we can check if it is an internal drop for onRemove
+    let onDrop = (e) => {
+      if (ref?.current && ref.current.contains(e.target)) {
+        setDroppedCollectionRef(ref);
+        setDroppedTarget(state.target);
+      }
+    };
+
+    document.addEventListener('drop', onDrop, true);
+    return () => {
+      document.removeEventListener('drop', onDrop, true);
```

Comment:
Current event flow is as follows: `dragstart` -> `drop` (provided onDrop doesn't fire yet due to setTimeout in useDrop) -> `dragend` (onDragEnd fires) -> `onDrop`. I need to set `droppedCollectionRef` and `droppedTarget` in the global state before `dragend` hence this capturing listener. 

Alternative would be to delay `options.onDragEnd` in `useDrag` similarly to how it is done in `useDrop` but that feels so iffy. Most ideal case is that browsers properly still call `ondrag`even if drop removes the original dragged elements